### PR TITLE
Syslog Provider Implementation

### PR DESCRIPTION
**Pull request by djhamb**
Implemented Syslog Provider using td-agent.

### CODE REVIEW

File: Docker/master/Dockerfile [hunk 1]
```diff
@@ -25,8 +32,8 @@ ENV LD_LIBRARY_PATH=/var/opt/microsoft/sapmon/${RELEASE}/sapmon/usrlibs/nwrfcsdk
 RUN curl --location --output pynwrfc-2.3.0-cp37-cp37m-linux_x86_64.whl https://github.com/SAP/PyRFC/releases/download/2.3.0/pynwrfc-2.3.0-cp37-cp37m-linux_x86_64.whl
 RUN pip3 install pynwrfc-2.3.0-cp37-cp37m-linux_x86_64.whl
 
-RUN ACCEPT_EULA=Y apt-get install msodbcsql17 -y
+# RUN ACCEPT_EULA=Y apt-get install msodbcsql17 -y
```
Comment (persiaAziz) [inline Docker/master/Dockerfile:35]: Why is this being commented out? We need this for sql provider

File: sapmon/content/Syslog.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
+{
+    "checks": [
+        {
+            "name": "Placeholder Test",
+            "description": "No-op",
```
Comment (persiaAziz) [inline sapmon/content/Syslog.json:5]: You can still put a one-line description of syslog provider

File: sapmon/payload/provider/syslog.py [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+                            customer_id {}
+                            shared_key {}
+                            log_type SyslogFromFluent
+                        </match>""".format(self.customerId, self.sharedKey)
+
+            config_file.write(content)
+        return True
+
+    # Restart td-agent service so that changes to config file take effect
+    def restartTdAgent(self) -> bool:
```
Comment (persiaAziz) [inline sapmon/payload/provider/syslog.py:74]: We will need to add error handling here

**Review by persiaAziz — CHANGES_REQUESTED** [review]

File: sapmon/payload/provider/syslog.py [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+        # update td-agent config
+        self.updateConfig()
+        
+        # restart td-agent service
+        if not self.restartTdAgent():
+            raise Exception("Failed to restart td-agent service")
+
+    # Update td-agent.conf file to send syslogs to log analytics
+    def updateConfig(self) -> bool:
+        with open(TD_AGENT_CONFIG_PATH, "w") as config_file:
```
Comment (persiaAziz) [inline sapmon/payload/provider/syslog.py:56]: please add error handler  here as well

File: sapmon/payload/provider/syslog.py [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+                            log_type SyslogFromFluent
+                        </match>""".format(self.customerId, self.sharedKey)
+
+            config_file.write(content)
+        return True
+
+    # Restart td-agent service so that changes to config file take effect
+    def restartTdAgent(self) -> bool:
+        try:
+            subprocess.run(['service', 'td-agent', 'restart'])
```
Comment (PakDLiu) [inline sapmon/payload/provider/syslog.py:78]: Looks like you are calling restart when the provider is initialized, does that mean td-agent is already started when the container starts? or will the restart command actually start td-agent?

Comment (djhamb) [reply sapmon/payload/provider/syslog.py:78]: Td-agent is already started when the container starts. The restart command is necessary to allow the changes to the td-agent config file to take effect. Any time the config is modified, the service has to be restarted.

**Review by PakDLiu — APPROVED** [review]

**Review by persiaAziz — APPROVED** [review]

**Pull request merged**